docs/contributor/arch/028-network-policy-hardening.md

```diff
@@ -0,0 +1,242 @@
+   - **Ingress Rules:**
+     - From: Any IP<br>
+       Ports: 9090
+   - **Egress Rules:**
+     - To: Any IP<br>
+       Ports: Any
+
+8. **Telemetry Manager**
+   - **Network Policy Name:** `telemetry-manager-manager`
+   - **Ingress Rules:**
+     - From: Any IP<br>
+       Ports: 8080, 8081, 9443
+   - **Egress Rules:**
+     - To: Any IP<br>
+       Ports: Any
+
+## Decision
+
+The current network policies are too weak. They do not meet the requirements described in https://github.com/kyma-project/kyma/issues/18818. These policies allow all IP addresses for both incoming and outgoing traffic. They only limit ports for incoming traffic, which means they can be made stronger. However, we will still need to allow any IP address (0.0.0.0/0) in some cases for incoming and outgoing rules. External services like CLS and Dynatrace, as well as Kube API server, use different IP address ranges that we cannot predict. This makes it hard to restrict outgoing traffic by IP address. Instead, we will restrict outgoing traffic by port number.
+
+We also decided to use `networking.kyma-project.io/metrics-scraping: allowed` label selector not only for RMA, but also for metric agent, self-monitoring, and customer-managed Prometheus deployments. Gardener system pods cannot be labeled in the zero-trust mode, so these pods must either be excluded from scraping.
+
+It is important to test the network policies using our E2E tests to ensure they function as intended. The problem is that k3s uses Flannel as the default CNI. Real Kyma clusters typically use Calico, which behaves slightly differently. We need to find a way to run our E2E tests with Calico to accurately validate the network policies.
+
+### What to do?
+
+# Phase 1: Hardening Existing Network Policies
+
+- Rename existing network policies to follow new naming conventions: `kyma-project.io--telemetry-<network-policy-name>`
+- Remove health check ports from ingress rules since they have no impact.
+- Remove Gardener system pods from our scraping jobs.
+- Implement cross-component network policies to allow essential services like DNS and Kubernetes API access.
+- Harden telemetry manager and self-monitoring since it requires no breaking changes.
+- Either expand Gardener E2E test suite to cover more scenarios or find a way to run E2E tests with Calico CNI.
+- Separate self-monitoring webhook from admission webhooks in telemetry manager to allow more fine-grained ingress rules.
+
+# Phase 2: Introduce Zero-trust Network Policies
+
+- Implement a feature toggle in the Telemetry CR to enable/disable extra rules that harden customer-to-telemetry communication as well as RMA, cross-Kyma module communication. The telemetry module already has basic network policies in place, making the `enableNetworkPolicies` toggle name not logical for our use case (we still will have network policies even if it is set to false). However, we will maintain this toggle for consistency with other Kyma modules.
```

Comment:
```suggestion
- Implement a feature toggle in the Telemetry CR to enable/disable extra rules that harden customer-to-telemetry communication as well as RMA, cross-Kyma module communication. Because the Telemetry module already has basic network policies in place, it's illogical to use the toggle name  **enableNetworkPolicies** (because we'll still have network policies even if it's set to `false`). However, we will maintain this toggle for consistency with other Kyma modules.
```

So we'll use the illogical toggle name anyway?